File: src/main/java/eu/karelhovorka/zpevnik/formatter/JMustacheTemplateSongFormatter.java
package eu.karelhovorka.zpevnik.formatter;


import com.samskivert.mustache.Mustache;

import eu.karelhovorka.zpevnik.text.SongText;

public class JMustacheTemplateSongFormatter extends MustacheTemplateSongFormatter {

    protected JMustacheTemplateSongFormatter(String mainTemplate, String chordReplaceTemplate) {
        super(mainTemplate, chordReplaceTemplate);
    }

    public static MustacheTemplateSongFormatter fromTemplates(String mainTemplate, String chordReplaceTemplate) {
        return new JMustacheTemplateSongFormatter(mainTemplate, chordReplaceTemplate);
    }

    @Override
    protected String applyTemplate(String template, SongText context) {
        //TODO: pre-compile and reuse template
        return Mustache.compiler().compile(template).execute(context);
    }
}

File: src/main/kotlinjvm/eu/karelhovorka/zpevnik/formatter/JMustacheTemplateSongFormatter.kt
package eu.karelhovorka.zpevnik.formatter


import com.samskivert.mustache.Mustache

import eu.karelhovorka.zpevnik.text.SongText

class JMustacheTemplateSongFormatter protected constructor(mainTemplate: String, chordReplaceTemplate: String) : MustacheTemplateSongFormatter(mainTemplate, chordReplaceTemplate) {

    override fun applyTemplate(template: String, context: SongText): String {
        //TODO: pre-compile and reuse template
        return Mustache.compiler().escapeHTML(false).compile(template).execute(context)
    }

    companion object {

        fun fromTemplates(mainTemplate: String, chordReplaceTemplate: String): MustacheTemplateSongFormatter {
            return JMustacheTemplateSongFormatter(mainTemplate, chordReplaceTemplate)
        }
    }
}
